velox/docs/develop/spilling.rst

```diff
@@ -199,23 +232,32 @@ small spill files as it might overload the metadata service of the storage
 system. On the other hand, we also want a sufficient number of spill files to
 parallelize the restore work. For example, to build a hash table from a
 spilled partition, we can parallelize the build work among multiple hash build
-operators by assigning each of them a shard of spill files.
-
-Given above considerations, we might just need to have a query config limit to
-cap a spill file size in practice. For unsorted spill, as we can continuously
-append to the same spill file so this helps to prevent a spill file from
-growing too big. For sorted spill, each file stores only one sorted run of
-data, hence, the spill file size is the minimum of spillable data size and
-query config limit. Configuration property kMaxSpillFileSize sets the spill
-file limit which can be tuned based on IO characteristics of the underlying
-storage system. We don’t expect it needs too much tuning in practice.
+operators by assigning each of them a shard of spill files. Given above
+considerations, we have the following two configuration properties to control
+the spill file size:
+
+**kMaxSpillFileSize** sets the maximum spill file size limit. For unsorted spill,
+as we continuously append to the same spill file so this helps to prevent a
+spill file from growing too big. For sorted spill, each file stores only one
+sorted run of data, hence, the spill file size is the minimum of spillable data
+size and this configuration limit.
+
+**kMinSpillRunSize** sets the minimum data size used by sorted spill to select
+partitions for spilling. Each sorted spill file can only store one sorted run
+of data. Spiller tries to spill from the same set of partitions if possible.
+By having this configuration limit, we can avoid spilling from partitions which
+have small amount of data, to avoid generating too many small spill files.
+
+Both configuration properties can be tuned based on IO characteristics of the
+underlying storage system. We don’t expect they needs too much tuning in
+practice.
 
 Spill Target Size
 ^^^^^^^^^^^^^^^^^
 The spill target size determines how much data to spill each time. If too
 small, spilling interrupts operator execution frequently and generates lots of
 small files. If too large, operator execution slows down by spilling lots of
-data to disk. Configuration property `kSpillableReservationGrowthPct <https://github.com/facebookincubator/velox/blob/main/velox/core/QueryConfig.h#L160>`_ sets the
+data to disk. Configuration property kSpillableReservationGrowthPct sets the
```

Comment:
kSpillableReservationGrowthPct -> spillable-reservation-growth-pct

Can we add this property to config.rst as well?

Reply:
Sure.